### PR TITLE
JDP200202-21 ProductController implementation

### PR DESCRIPTION
**Pull request by TomaszTomasiak**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/domain/dto/ProductDto.java [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
 public class ProductDto {
-}
+    private Long id;
+    private String name;
+    private BigDecimal price;
+    private String description;
+    private Group groupId;
```
Comment (TomaszTomasiak) [inline src/main/java/com/kodilla/ecommercee/domain/dto/ProductDto.java]: tutaj będę zmieniał na private Long groupId;

File: src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java [hunk 2]
```diff
@@ -1,41 +1,40 @@
...
-                loanDto.getReturnedDate());
+    ProductRepository productRepository;
+
+    public Product mapToProduct(final ProductDto productDto) {
+        return new Product(
+                productDto.getId(),
+                productDto.getName(),
+                productDto.getPrice(),
+                productDto.getDescription(),
+                productDto.getGroupId());
```
Comment (TomaszTomasiak) [inline src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java]: będzie zmiana typu  groupId z Group na Long

File: .gitignore [hunk 3]
```diff
@@ -23,4 +23,7 @@
 /nbbuild/
 /dist/
 /nbdist/
-/.nb-gradle/
\ No newline at end of file
+/.nb-gradle/
+
+**/out/
+**/build/
```
Comment (g-chmura) [inline .gitignore]: No new line at the end of the file

File: src/main/java/com/kodilla/ecommercee/domain/dto/ProductDto.java [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class ProductDto {
+    private Long id;
+    private String name;
+    private BigDecimal price;
+    private String description;
+    private Group groupId;
+}
```
Comment (g-chmura) [inline src/main/java/com/kodilla/ecommercee/domain/dto/ProductDto.java]: No newline at the end of the file

**Comment by TomaszTomasiak** [comment]
> Proponowałbym dodać atrybut value:
> 
> `@RestController
> @RequestMapping("api/v1/ecommercee/products")
> public class ProductController {
> 
> ```
> @Autowired
> ProductService productService;
> 
> @Autowired
> ProductMapper productMapper;
> 
> @GetMapping(value = "/all", consumes = MediaType.APPLICATION_JSON_VALUE)
> public List<ProductDto> getAllProducts() {
>     return productMapper.mapToProductDtoList(productService.getProducts());
> }
> 
> @GetMapping(value = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
> public ProductDto getProduct(@PathVariable("id") Long id) throws NotFoundException {
>     return productMapper.mapToProductDto(productService.getProduct(id).orElseThrow(NotFoundException::new));
> }
> 
> @PostMapping(value = "new", consumes = MediaType.APPLICATION_JSON_VALUE)
> public void createProduct(@RequestBody ProductDto productDto) {
>     productService.saveProduct(productMapper.mapToProduct(productDto));
> }
> 
> @PutMapping(value = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
> public ProductDto updateProduct(@PathVariable Long id, @RequestBody(required = false) ProductDto productDto) {
>     return productMapper.mapToProductDto(productService.saveProduct(productMapper.mapToProduct(productDto)));
> }
> 
> @DeleteMapping(path = "/{id}")
> public void deleteProduct(@PathVariable Long productId) {
>     productService.deleteProduct(productId);
> }
> ```
> 
> }`
> 
> W tym ostatnim to też w sumie mogłoby być value (z tego co przeczytałem oba robią to samo).
> 
> Można też pododawać try/catch do wszystkich metod poza pierwszą (która i tak zawsze coś zwróci).

**Pull request closed without merging**